internal/session/handle_search.go

```diff
@@ -3,16 +3,16 @@ package session
 import (
 	"context"
 
+	"github.com/ProtonMail/gluon/imap/command"
 	"github.com/ProtonMail/gluon/internal/contexts"
-	"github.com/ProtonMail/gluon/internal/parser/proto"
 	"github.com/ProtonMail/gluon/internal/response"
 	"github.com/ProtonMail/gluon/internal/state"
 	"github.com/ProtonMail/gluon/profiling"
 	"golang.org/x/text/encoding"
 	"golang.org/x/text/encoding/ianaindex"
 )
 
-func (s *Session) handleSearch(ctx context.Context, tag string, cmd *proto.Search, mailbox *state.Mailbox, ch chan response.Response) (response.Response, error) {
+func (s *Session) handleSearch(ctx context.Context, tag string, cmd *command.Search, mailbox *state.Mailbox, ch chan response.Response) (response.Response, error) {
```

Comment:
Haven't finished reviewing yet, but note to self: now that we should be able to support parsing arbitrary bytes, let's ensure we have a test for SEARCH with non-utf-8 stuff (which should have failed before but should pass now)

Reply:
There is GODT-2165 which covers this topic. We can work on that after this MR.

Reply:
Thing is, we had a test that would assert that non-utf-8 search would throw an error and that was deleted -- couldn't you just for this MR assert that no error is thrown? Then when we do GODT-2165 we can extend the test to make sure decoding works properly? Just wanting to make sure we don't get any surprises with this MR.

Reply:
This error no longer exists. It doesn't make sense anymore.

Reply:
Renabled disabled test covered by GODT-2165 to verify that decoding of non-ascii chars works.